From kaneshige.kenji@jp.fujitsu.com Wed Jan 25 17:05:23 2006
Message-ID: <43D81FEC.1040804@jp.fujitsu.com>
Date: Thu, 26 Jan 2006 10:03:40 +0900
From: Kenji Kaneshige <kaneshige.kenji@jp.fujitsu.com>
To: Greg KH <gregkh@suse.de>
Cc: Kenji Kaneshige <kaneshige.kenji@jp.fujitsu.com>
Subject: [PATCH 9/11] shpchp -  removed unncessary 'magic' member from slot

This patch removes unnecessary 'magic' member from struct slot of
SHPCHP driver.

Signed-off-by: Kenji Kaneshige <kaneshige.kenji@jp.fujitsu.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/shpchp.h      |    6 ------
 drivers/pci/hotplug/shpchp_core.c |    1 -
 2 files changed, 7 deletions(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/shpchp.h
+++ gregkh-2.6/drivers/pci/hotplug/shpchp.h
@@ -53,9 +53,7 @@ extern int shpchp_debug;
 #define info(format, arg...) printk(KERN_INFO "%s: " format, MY_NAME , ## arg)
 #define warn(format, arg...) printk(KERN_WARNING "%s: " format, MY_NAME , ## arg)
 
-#define SLOT_MAGIC	0x67267321
 struct slot {
-	u32 magic;
 	u8 bus;
 	u8 device;
 	u16 status;
@@ -287,10 +285,6 @@ static inline int slot_paranoia_check (s
 		dbg("%s - slot == NULL", function);
 		return -1;
 	}
-	if (slot->magic != SLOT_MAGIC) {
-		dbg("%s - bad magic number for slot", function);
-		return -1;
-	}
 	if (!slot->hotplug_slot) {
 		dbg("%s - slot->hotplug_slot == NULL!", function);
 		return -1;
--- gregkh-2.6.orig/drivers/pci/hotplug/shpchp_core.c
+++ gregkh-2.6/drivers/pci/hotplug/shpchp_core.c
@@ -137,7 +137,6 @@ static int init_slots(struct controller 
 		hotplug_slot->name = name;
 
 		slot->hp_slot = i;
-		slot->magic = SLOT_MAGIC;
 		slot->ctrl = ctrl;
 		slot->bus = ctrl->slot_bus;
 		slot->device = ctrl->slot_device_offset + i;
